Topics/Development_Process/QA_testing.md

```diff
@@ -0,0 +1,51 @@
+# QA Testing: What is it? Why should we do it? How do we do it?
+
+## Table of Contents:
+### [Introduction](#introduction-1)
+### [What is Software QA?](#what-is-software-qa-1)
+### [Why should we test?](#why-should-we-test-1)
+### [How to write test cases?](#how-to-write-test-cases-1)
+
+
+## Introduction
+
+This article will specifically focus on the testing portion of the development process. It is an essential and often overlooked part of this process and it will explain what software quality assurance or Software QA is, why it is an important part of the development process and how to do it.
+At the end of this article, a CSC301 student would have the basic understandings of Software QA and would be able to look into a career in the Software QA field.
+
+
+## What is Software QA?
+
+[Quality Assurance, Quality Control and Testing — the Basics of Software Quality Management](https://www.altexsoft.com/whitepapers/quality-assurance-quality-control-and-testing-the-basics-of-software-quality-management/)
+
+This paper is likely the greatest paper about software QA, however it is quite long. Here are some of the main definitions and key takeaways:
+- In order to make sure the released software is safe and functions as expected, the concept of software quality was introduced. It is often defined as “the degree of conformance to explicit or implicit requirements and expectations”.
+- The purpose of QA is to set up adequate processes, introducing the standards of quality to prevent the errors and flaws in the product. This is slightly different to testing where the purpose is to detect and solve software errors and flaws.
+- There are different levels of software testing, component/unit testing, integration testing, system testing then acceptance testing.
+
+
+This article gives a tease of what a software QA engineer is like, the different types of SQA engineers and their duties and responsibilities. It is the right article to figure out if you want a career in this field or you are interested in QA.
+[What does a Software Quality assurance engineer do?](https://www.careerexplorer.com/careers/software-quality-assurance-engineer/)
```

Comment:
I was wondering if you meant to show the link to the article first before the description?

Reply:
I meant to put it after. Do you think it should go before?

Reply:
Oh I see. I think it could go either way. I assumed it would be before since the previous paper on line 19 was added before the explanation.

Reply:
You are right. Best to be consistent. I made that change.